EIPS/eip-1716.md

```diff
@@ -0,0 +1,31 @@
+---
+eip: 1716
+title: "Hardfork Meta: Petersburg"
+author: Afri Schoedon (@5chdn)
+type: Meta
+status: Draft
+created: 2019-01-21
+requires: 1013, 1283
+---
+
+## Abstract
+
+This meta-EIP specifies the changes included in the Ethereum hardfork that removes [EIP-1283](./eip-1283.md) from [Constantinople](./eip-1013.md).
+
+## Specification
+
+- Codename: Petersburg
+- Aliases: St. Petersfork, Peter's Fork, Constantinople Fix
+- Activation:
+  - `Block >= 7_280_000` on the Ethereum mainnet
+  - `Block >= 4_939_394` on the Ropsten testnet
+- Removed EIPs:
+  - [EIP 1283](./eip-1283.md): Net gas metering for SSTORE without dirty maps
+
```

Comment:
Potential clarifications: 
* If `Petersburg` and `Constantinople` are applied at the same block, `Petersburg` takes precedence: with the net effect of EIP 1283 being _disabled_. 
* If `Petersburg` is defined with an earlier block number than `Constantinople`, then there is _no immediate effect_ from the `Petersburg` fork. However, when `Constantinople` is later activated, EIP-1283 should be _disabled_.

Reply:
I agree with @axic It will better if we can set Petersburg block number >= Constantinople

Reply:
It makes it unnecessary complicated. The clarification Martin proposed clearly states how a client should behave if Petersburg < Constantinople.

Reply:
I beg to differ. With allowing Petersburg before Constantinople the client must track a flag, while my proposal would allow the client to just reject starting after parsing the block numbers in the configuration.

Reply:
Well, the flag needs to be tracked regardless. With the proposed solution, we don't have to add an extra constraint when validating the configuration. It also makes it easier to manually/semi-manually create configurations, since you can set the `petersburg` block to `0` and don't worry about moving that one around if you move constantinople. 
Also, geth has `override.constantinople` flag, which would continue to work nicely (and I'd recommend Parity to implement it too). If we change like you propose, the user would have to override both.

Reply:
Wouldn't it be better to specify it that Petersburg activates EIP 145, 1014, 1052, 1234 and disables 1283?

Reply:
No, Constantinople enables EIP 145, 1014, 1052, 1234, and 1283; Petersfork disables 1283. I don't see why we should have two hardforks for the same thing.